pkg/api/server/register_registries.go

```diff
@@ -0,0 +1,37 @@
+package server
+
+import (
+	"net/http"
+
+	"github.com/containers/libpod/pkg/api/handlers/libpod"
+	"github.com/gorilla/mux"
+)
+
+func (s *APIServer) registerRegistriesHandlers(r *mux.Router) error {
+	// swagger:operation GET /libpod/registries/json libpod libpodListRegistries
+	// ---
+	// summary: Return a list of search registries and their mirrors.
+	// description: Return a sorted list of all search registries and their mirrors. The list can be used to determine which credentials the server may require.
```

Comment:
Sort criteria?

Reply:
I really want to ditch this endpoint in favour of sending all credentials (see https://github.com/containers/image/pull/942#issue-422757846).

Getting this to work is exploding in complexity and amount of work, so I hope we can settle on the all-credentials case.